### PR TITLE
Tech CORE-492 | Correction on the controller operation

### PR DESCRIPTION
**Pull request by LucasStarlingdePaulaSalles**
# Description

Kind: Tech

Section: Insprd Controller

Summary:
The objective here was to verify that the scope field from controller dapps actually had effect on its operation. It turns out it works like it is supposed to, however the exploration of this feature revealed bugs regarding the operation of controller dapps. The operation for apps that were compound of several nodes was failing because the k8s operator was operating on the permanent memory instead of its updated copy. To fix this a couple adjustments were made and it is now working properly.

### CODE REVIEW

File: cmd/insprd/operators/nodes/converter.go [hunk 1]
```diff
@@ -179,17 +196,34 @@ func (no *NodeOperator) withBoundary(app *meta.App) k8s.ContainerOption {
...
+			var cterr, cherr error
+			if usePermTree {
+				ch, cherr = no.memory.Perm().Channels().Get(parent, chName)
+				ct, cterr = no.memory.Perm().Types().Get(parent, ch.Spec.Type)
+			} else {
+				ch, cherr = no.memory.Channels().Get(parent, chName)
+				ct, cterr = no.memory.Types().Get(parent, ch.Spec.Type)
+			}
+			if cherr != nil {
+				logger.Error("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa")
```
Comment (luiz-couto) [inline cmd/insprd/operators/nodes/converter.go]: remove this aaaaaaaaaaaaaaaaaaaaaaaa log

File: cmd/insprd/operators/nodes/converter.go [hunk 2]
```diff
@@ -179,17 +196,34 @@ func (no *NodeOperator) withBoundary(app *meta.App) k8s.ContainerOption {
...
+				panic(err)
+			}
+			if cterr != nil {
+				logger.Error("taaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa")
```
Comment (luiz-couto) [inline cmd/insprd/operators/nodes/converter.go]: this too

**Review by luiz-couto — APPROVED** [review]

**Review by lucasABLima — APPROVED** [review]

File: cmd/insprd/memory/tree/channel.go [hunk 3]
```diff
@@ -33,7 +33,7 @@ func (chh *ChannelMemoryManager) Get(scope, name string) (*meta.Channel, error)
 		zap.String("channel", name),
 		zap.String("scope", scope),
 	)
-	l.Debug("recevide channel retrieval request")
+	l.Debug("recevid channel retrieval request")
```
Comment (nicholaspcr) [inline cmd/insprd/memory/tree/channel.go:36]: `received` instead of `recevid`, no?

File: cmd/insprd/operators/nodes/converter.go [hunk 4]
```diff
@@ -179,17 +196,34 @@ func (no *NodeOperator) withBoundary(app *meta.App) k8s.ContainerOption {
...
+			if usePermTree {
+				ch, cherr = no.memory.Perm().Channels().Get(parent, chName)
+				ct, cterr = no.memory.Perm().Types().Get(parent, ch.Spec.Type)
+			} else {
+				ch, cherr = no.memory.Channels().Get(parent, chName)
+				ct, cterr = no.memory.Types().Get(parent, ch.Spec.Type)
+			}
+			if cherr != nil {
+				logger.Error("Unable to get channel to resolve with boundary", zap.String("channel", chName))
+				panic(err)
```
Comment (nicholaspcr) [inline cmd/insprd/operators/nodes/converter.go:215]: couldn't you do something like

```go
var mem "interface that Perm and memory return"
if usePermTree{
    mem = no.memory.Perm()
else {
    mem = no.memory
}

//  the rest stays liike
ch, cherr := mem.Channels()
ct, cterr := mem.Types()
```

File: cmd/insprd/operators/nodes/converter_test.go [hunk 5]
```diff
@@ -96,7 +96,7 @@ func Test_intToint32(t *testing.T) {
 	}
 }
 
-func TestNodeOperator_withBoundary(t *testing.T) {
+func TestNodeOperator_withBoundary(t *testing.T) { // needs more testing, longer creates, updates, and deletes
```
Comment (nicholaspcr) [inline cmd/insprd/operators/nodes/converter_test.go:99]: add a `TODO` then, easier to go around the project searching for a specific tag than to find this comment

File: cmd/insprd/operators/nodes/converter.go (hunk 4 above)
Comment (LucasStarlingdePaulaSalles) [reply cmd/insprd/operators/nodes/converter.go:215]: not the same interface

**Pull request merged**